src/WebJobs.Script/Description/Proxy/ProxyFunctionInvoker.cs

```diff
@@ -0,0 +1,27 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.Azure.AppService.Proxy.Client.Contract;
+
+namespace Microsoft.Azure.WebJobs.Script.Description
+{
+    internal class ProxyFunctionInvoker : FunctionInvokerBase
+    {
+        private IProxyClient _proxyClient;
+
+        public ProxyFunctionInvoker(ScriptHost host, FunctionMetadata functionMetadata, IProxyClient proxyClient) : base(host, functionMetadata)
+        {
+            _proxyClient = proxyClient;
+        }
+
+        protected override async Task InvokeCore(object[] parameters, FunctionInvocationContext context)
+        {
+            await _proxyClient.CallAsync(parameters, null, context.Logger);
```

Comment:
The interface is kept loosely coupled so that we do not need a major version change to introduce new parameters. Major version change needs customer action hence its best avoided.

For e.g. Lets say we need to add a new parameter for some reason. Then we would first add support for this in the client dlls, the Shim would not need any changes. once its deployed, we can then do the required changes in functions and do the function deployment. This keeps funcitons and proxies deployment decoupled. If we do a strong signature, we would need to have a new major version of gibraltar which would need customer action, or have dependency on deployments where function and proxies are deployed as together.

---
In reply to: [133089988](https://github.com/Azure/azure-webjobs-sdk-script/pull/1749#discussion_r133089988) [](ancestors = 133089988)